src/v/hashing/tests/secure_tests.cc

```diff
@@ -0,0 +1,31 @@
+// Copyright 2020 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#define BOOST_TEST_MODULE secure_hash
+#include "hashing/secure.h"
+
+#include <boost/test/unit_test.hpp>
+
+template<typename hmac>
+static auto hmac_test_vector() {
+    hmac mac("redpanda");
+    mac.update("is the cutest panda");
+    return mac.reset();
+}
+
+BOOST_AUTO_TEST_CASE(hmac) {
+    BOOST_REQUIRE_EQUAL(
+      to_hex(hmac_test_vector<hmac_sha256>()),
+      "89fd20823dc7f76e21feeb350f5f08f7fc097f82709cb4212cc820353c589e12");
```

Comment:
did you verify these hex on the command line or somewhere else too.

Reply:
pretty sure i used generated this test cases using hmac cli tools on fedora.

Reply:
lgtm. would be nice to record the command line as a comment.

Reply:
done